### PR TITLE
Development

### PR DESCRIPTION
**Pull request by belfortep**
First implementation of Conway's game logic and tests

### CODE REVIEW

File: src/conways/point.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+use std::collections::HashSet;
+
+#[derive(PartialEq, Eq, Hash, Clone, Copy)]
+pub struct Point {
+    x_position: i32,
+    y_position: i32,
+}
+
+impl Point {
+    pub fn new(x_position: i32, y_position: i32) -> Self {
```
Comment (lazcanoluca) [inline src/conways/point.rs:10]: I don't get what is the frame of reference for the creation of cells. Intuitively, I would think of the cell position as an xy coordinate in the first region (x and y both positive), but x and y are defined as `i32`, and when defining the cells you use both positive and negative coordinates. I don't understand the rationale. Is there something I am missing?

File: src/conways/conways_game.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
+use std::collections::HashSet;
+
+use super::point::Point;
+
+pub struct ConwaysGame {
+    cells: HashSet<Point>,
+}
+
+impl ConwaysGame {
+    pub fn new(cells: Vec<Point>) -> Self {
```
Comment (lazcanoluca) [inline src/conways/conways_game.rs]: I like the `Point` entity, it makes the positioning clear. Now, I can't infer if when creating a `ConwaysGame` the cells are initialized as dead or alive, or if the vector of points is the vector of alive cells.

**Review by lazcanoluca — COMMENTED** [review]
Looks good as a starting point! I left a couple of comments asking for clarifications.

File: src/conways/conways_game.rs (hunk 2 above)
Comment (belfortep) [reply src/conways/conways_game.rs]: Yes, you are right, i'm changing the variable to a more descriptive name

File: src/conways/point.rs (hunk 1 above)
Comment (belfortep) [reply src/conways/point.rs:10]: With the implementation using `HashSet` the points could be anywhere in the xy plane, so it doesn't really matter if the positions are negative or positive

Comment (lazcanoluca) [reply src/conways/point.rs:10]: I see. I think it's simpler to restrict the coordinates to a single quadrant. Don't worry, you can mark this one as resolved.

**Comment by lazcanoluca** [comment]
🚀 🚀

**Review by lazcanoluca — APPROVED** [review]
🚀 🚀

**Pull request merged**